src/main/src/main/java/org/geoserver/catalog/CoverageViewReader.java

```diff
@@ -284,35 +292,115 @@ public GridCoverage2D read(GeneralParameterValue[] parameters) throws IllegalArg
         List<CoverageBand> bands = coverageView.getCoverageBands();
         List<GridCoverage2D> coverages = new ArrayList<GridCoverage2D>();
         List<SampleDimension> dims = new ArrayList<SampleDimension>();
-
-        // Use composition rule specific implementation
+        
         CoveragesConsistencyChecker checker = null;
-        for (CoverageBand band : bands) {
-            // Refactor this once supporting complex compositions
-            final InputCoverageBand inputBand = band.getInputCoverageBands().get(0); 
-            final String coverageName = inputBand.getCoverageName();
-            final GridCoverage2DReader reader = wrap(delegate, coverageName, coverageInfo);
-
-            // Remove this when removing constraints
-            if (checker == null) {
-                checker = new CoveragesConsistencyChecker(reader);
-            } else {
-                checker.checkConsistency(reader);
+        
+        int coverageBandsSize = bands.size();
+        
+        //Check params, populate band indices to read if BANDS param has been defined
+        ArrayList<Integer> selectedBandIndices = new ArrayList<Integer>();
+    	for (int m=0;m<coverageBandsSize;m++)
+    		selectedBandIndices.add(m);
```

Comment:
There are tabs in the code, please follow the coding conventions: http://docs.geoserver.org/latest/en/developer/eclipse-guide/#code-formatting